### ISSUE TITLE
'./please tools update-dependencies' generates broken nix files

### ISSUE THREAD

**Issue by marco-c** [mozilla-releng/services#872]
After I update the dependencies for shipit-uplift and then try to run `./please check shipit-uplift`, I get:
```
error: infinite recursion encountered, at undefined position
(use ‘--show-trace’ to show detailed location information)
```

**Comment by marco-c** [comment]
Here's the generated diff:
[diff.zip](https://github.com/mozilla-releng/services/files/1759291/diff.zip)

**Comment by garbas** [comment]
@marco-c in python cyclic dependencies are allowed and in nix this is now allowed. i'm just about to release new version of pypi2nix which might solve some of our problems also for the future. but we can improve this later.

the cycling becomes bewteen `attrs` and `pytest`. you need to remove one or the other from others `propagatedBuildInputs` by overriding the derivation in `nix/requirements_override.nix`.

We can look at this together if you want, or I just update shipit-uplift it myself.

I think it would be worthwhile to run this update of dependencies on a weekly basis. that way i can catch this error ahead of time and developers don't really need to care about them.

**Comment by marco-c** [comment]
> We can look at this together if you want, or I just update shipit-uplift it myself.

Yes, please. If you have time tomorrow I'd like to see how it can be done.

> I think it would be worthwhile to run this update of dependencies on a weekly basis. that way i can catch this error ahead of time and developers don't really need to care about them.

Makes perfect sense to me.

**Comment by marco-c** [comment]
Fixed by #875. Thanks for the help @garbas.
